Makefile

```diff
@@ -90,10 +96,22 @@ misspell:
 build:
 	go build -o ./bin/$(OTELCOL_BINARY) $(LD_FLAGS) ./cmd/otelopscol
 
+.PHONY: build_gpu_support
+build_gpu_support:
+	go build -tags=gpu -o ./bin/$(OTELCOL_BINARY) $(LD_FLAGS) ./cmd/otelopscol
```

Comment:
I have a suggestion here: 
Instead of making targets for each combination of the tags, we could use a variable:
Remove `build_gpu_support` target and modify the original `build` target:
```
build:
	go build -tag=$(GO_TAGS) -o ./bin/$(OTELCOL_BINARY) $(LD_FLAGS) ./cmd/otelopscol
```
and when calling we can do `make build` for gpu off, and `make GO_TAGS=gpu` for gpu on. And we can do this for the testing targets as well. This may be beneficial if we have more tags, but how to run with `gpu` would become less obvious. 

I don't have a strong preference between those two, so I will ask you to decide!

Reply:
Done in [f3be5f6](https://github.com/GoogleCloudPlatform/opentelemetry-operations-collector/pull/156/commits/f3be5f62d83b33c2a8aecef71ad73eb27211f5ef)